### ISSUE TITLE
Add mock for eth_requestAccounts

### ISSUE THREAD

**Issue by code3z** [VGLoic/eth-testing#10]

**Comment by VGLoic** [comment]
Hey @code3z ! Thanks again for this another issue :)

The `eth_requestAccounts` is very particular to MetaMask, I was not sure if it was worth it to add a high level utils dedicated to it 🤔. The current alternative is to use the `mockRequest` in the `lowLevel` object directly as 
```ts
testingUtils.lowLevel.mockRequest("eth_requestAccounts", [address]);
```

An example can be found [here](https://github.com/VGLoic/metamask-react/blob/main/src/__tests__/index.test.tsx).

The options are basically
- add a dedicated part to the README with the lowLevel mockRequest example
- add a high level util dedicated to `eth_requestAccounts` of MetaMask

Any strong opinion on your side? :) It would help me take a direction path!

**Comment by VGLoic** [comment]
Good stuff, thanks a lot! Let's go for the high level utils then!

So I started a [PR](https://github.com/VGLoic/eth-testing/pull/12) for that. It is very simple for now, it is basically only a wrapper around the low level method.

I'm a bit afraid it does not really solve your case, so if you have any feedbacks I would gladly take it :D

One step forward that would be interesting to add IMO is the automatic mock of the `eth_accounts` **only after** the `eth_requestAccounts` has resolved. An example of what it could be solved would be
```
// Start the test saying that no accounts are connected
testingUtils.mockAccounts([]);
// Mock the next eth_requestAccounts request
testingUtils.mockRequestAccounts(["0xf61B443A155b07D2b2cAeA2d99715dC84E839EEf"]);
// Once the eth_requestAccounts has resolved, eth_accounts is still mocked to return [] which is problematic
```

**Comment by code3z** [comment]
Looks good.

> One step forward that would be interesting to add IMO is the automatic mock of the `eth_accounts` **only after** the `eth_requestAccounts` has resolved. An example of what it could be solved would be

yes, this sounds great!

**Comment by VGLoic** [comment]
Hey @code3z! Just telling you that I have completed the [PR](https://github.com/VGLoic/eth-testing/pull/12) in order to have this automatic mock once the `eth_requestAccounts` has been consumed, e.g.
```ts
       // Start with no accounts - the wallet is not connected
        testingUtils.mockAccounts([]);

        // After the eth_requestAccounts has resolved
        // - the account will be "0xf61B443A155b07D2b2cAeA2d99715dC84E839EEf",
        // - the chain will be "0x1",
        testingUtils.mockRequestAccounts(
            ["0xf61B443A155b07D2b2cAeA2d99715dC84E839EEf"],
            { chainId: "0x1" }
        );
        ...
```
And in general, trigger callbacks have been added to the mocks (using low level utils)

Feel free to comment if you have any feedbacks on this! :)
